File: algorithms/1260.shift-2d-grid/solution.rs
impl Solution {
    pub fn shift_grid(grid: Vec<Vec<i32>>, k: i32) -> Vec<Vec<i32>> {
        let mut horiz = grid.clone().into_iter().flatten().collect::<Vec<i32>>();
        let time = k as usize % horiz.len();
        for _ in 0..time {
            let tmp = horiz.pop().unwrap();
            horiz.insert(0, tmp);
        }

        horiz
            .chunks(grid[0].len())
            .into_iter()
            .map(|chunk| chunk.to_vec())
            .collect()
    }
}

File: algorithms/1260.shift-2d-grid/solution.rb
# @param {Integer[][]} grid
# @param {Integer} k
# @return {Integer[][]}
def shift_grid(grid, k)
  horiz = grid.flatten
  (k % horiz.size).times do
    tmp = horiz.pop
    horiz.insert(0, tmp)
  end

  horiz.each_slice(grid[0].size).to_a
end
